cpp/dolfinx/io/checkpointing.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (C) 2024 Abdullah Mujahid
+//
+// This file is part of DOLFINX (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#ifdef HAS_ADIOS2
+
+#include "checkpointing.h"
+#include "ADIOS2_utils.h"
+#include <adios2.h>
+#include <basix/finite-element.h>
+#include <dolfinx/mesh/Mesh.h>
+#include <mpi.h>
+
+using namespace dolfinx;
+using namespace dolfinx::io;
+
+/// @file checkpointing.h
+/// @brief ADIOS2 based checkpointing
+namespace
+{
+std::map<basix::element::lagrange_variant, std::string> lagrange_variants{
```

Comment:
Is there a reason not to place this in Basix?

Reply:
It was suggested to put this map where ever it is needed instead of placing it in Basix.
See https://github.com/FEniCS/basix/pull/839#issuecomment-2209534926